strategies/failed/src/main/java/org/arquillian/smart/testing/strategies/failed/InProjectTestReporterLoader.java

```diff
@@ -38,11 +47,12 @@
                         }
                     })
                     .map(is -> {
-                        try {
-                            return SurefireReader.loadTestResults(is);
-                        } catch (XMLStreamException e) {
-                            return null;
+                        final Optional<TestResultParser> testResultParser = javaSPILoader.onlyOne(TestResultParser.class, trp -> "junit".equals(trp.type()));
```

Comment:
Or let me ask you differently - let's assume Serenity BDD has different test reports and we would like to use them to know which tests are failing. How would I parse them in the `failed` strategy with this implementation?